### PR TITLE
Tracking NN including labels

### PR DESCRIPTION
**Pull request by pet1330**
Adds Maximum cost to labels which do not match.

### CODE REVIEW

File: include/bayes_tracking/multitracker.h [hunk 1]
```diff
@@ -230,6 +236,17 @@ class MultiTracker {
         m_filters[j].filter->predict_observation(om, zp, Zp);
         S = Zp + om.Z;  // H*P*H' + R
         for (int i = 0; i < M; i++) {
+
+         // Only Check NN, NNJPDA tag associate not yet implemented
+         if (alg == NN) 
+         {
+           // Assign maximum cost if observations and trajectories labelled do not match
+           if (m_observations[i].tag != m_filters[j].tag)
```
Comment (marc-hanheide) [inline include/bayes_tracking/multitracker.h]: This means that whenever `NN` is used, the tag checking will be employed, right? That's a change to the default behaviour that present to date. I think, it should be an optional behaviour. I suggest to actually implement this as another association algorithm, e.g. `NN_TAGGED` or `NN_LABELED` and making sure the original one is left as is.

Comment (cdondrup) [inline include/bayes_tracking/multitracker.h]: :+1:

**Comment by pet1330** [comment]
Yeah, I can change that. My thinking behind combining them was that if a label is provided then it is used to help distinguish them, but if a label is not provided then it is assigned an empty string, which when compared, will match and have no affect.

The only time this will ever differ from the original is when a label is provided but you don't want them to be further distinguished by it. Would this ever be something that you would want?

**Comment by pet1330** [comment]
Don't merge this yet. I've noticed some errors which need correcting first.

**Comment by pet1330** [comment]
That passed my tests, but needs to be checked :smile:

**Comment by pet1330** [comment]
retest this please

**Comment by pet1330** [comment]
@marc-hanheide can you merge this?

**Pull request merged**